cachelib/cachebench/workload/KVReplayGenerator.h

```diff
@@ -323,27 +389,91 @@ inline void KVReplayGenerator::genRequests() {
         req = std::make_unique<ReqWrapper>(*reqWrapper);
       }
 
+      size_t keySize = req->key_.size();
       if (ampFactor_ > 1) {
         // Replace the last 4 bytes with thread Id of 4 decimal chars. In doing
         // so, keep at least 10B from the key for uniqueness; 10B is the max
         // number of decimal digits for uint32_t which is used to encode the key
-        if (req->key_.size() > 10) {
+        if (keySize > 10) {
           // trunkcate the key
-          size_t newSize = std::max<size_t>(req->key_.size() - 4, 10u);
+          size_t newSize = std::max<size_t>(keySize - 4, 10u);
           req->key_.resize(newSize, '0');
         }
         req->key_.append(folly::sformat("{:04d}", keySuffix));
       }
 
-      auto shardId = getShard(req->req_.key);
-      auto& stressorCtx = getStressorCtx(shardId);
-      auto& reqQ = *stressorCtx.reqQueue_;
+      if (makeBinaryFile_) {
+        uint8_t op = 0;
+
+        switch (req->req_.getOp()) {
+        case OpType::kGet:
+          op = 1;
+          break;
+        case OpType::kSet:
+          op = 2;
+          break;
+        case OpType::kDel:
+          op = 3;
+          break;
+        }
+        auto binReq = BinaryRequest(keySize, req->sizes_[0], req->repeats_, op,
+                                    req->req_.ttlSecs, keyOffset);
+        std::memcpy(outputStreamReqs_ + nreqs, &binReq, sizeof(binReq));
+        std::memcpy(outputStreamKeys_ + keyOffset, req->key_.c_str(), keySize);
+        if ((nreqs % BIN_REQ_INT) == 0 && nreqs > 0) {
```

Comment:
can you add a comment for this part of the code?

Reply:
done